vllm/worker/model_runner.py

```diff
@@ -504,27 +656,32 @@ def build(self) -> ModelInputForGPU:
         create on-device tensors.
         """
         # Combine and flatten intermediate data.
-        input_tokens = flatten_2d_lists([
-            flatten_2d_lists(inter_data.input_tokens)
-            for inter_data in self.inter_data_list
-        ])
+        input_tokens = []
+        for inter_data in self.inter_data_list:
+            for cur_input_tokens in inter_data.input_tokens:
+                input_tokens.extend(cur_input_tokens)
```

Comment:
The standard python way of doing this is:
```python
input_tokens = [
    in_toks for inter_data in self.inter_data_list
    for in_toks in inter_data.input_tokens
]
```

I'd guess that this would be more efficient but can never be sure, would have to microbench it...

Reply:
Yeah, this is what flatten_2d_lists(..) was doing, but extend() is faster.

Reply:
We should change flatten_2d_list to use extend then.

Reply:
The prior code wasn't doing the same thing. Technically this isn't flattening 2d lists - it's a list of objects each having a list `inter_tokens` field. So flatten_2d_lists() isn't appropriate here anyhow. If you look at the prior code is calling flatten_2d_lists twice, which is probably the reason for the excess allocations and slower speed.

Reply:
Hmm yeah you're right. I was hoping there's a way to extract this logic to be a utility function, but seems not trivial.

Reply:
I see